docs/data/data-grid/editing/editing.md

```diff
@@ -157,18 +157,61 @@ The object also contains a `reason` param that specifies which type of interacti
 The following demo shows how to prevent the user from exiting edit mode when clicking outside of a cell.
 To do this, the `onCellEditStop` prop is used to check if the `reason` is `'cellFocusOut'`.
 If that condition is true, it [disables](/x/react-data-grid/events/#disabling-the-default-behavior) the default event behavior.
-In this scenario, the user can only stop editing a cell by pressing <kbd class="key">Enter</kbd>, <kbd class="key">Escape</kbd> or <kbd class="key">Tab</kbd>.
+In this context, the user can only stop editing a cell by pressing <kbd class="key">Enter</kbd>, <kbd class="key">Escape</kbd> or <kbd class="key">Tab</kbd>.
 
 {{"demo": "DisableStopEditModeOnFocusOut.js", "bg": "inline"}}
 
-### Disabling default start and stop behavior
+## Controlled mode
 
-The following demo shows how external buttons can be used to start and stop edit mode.
-To do this, the default behavior of the `onCellEditXXX` events is disabled.
-To edit a cell, click on it, then click **Edit**.
+Each cell and row has two modes: `edit` and `view`.
+You can control the active mode using the props `cellModesModel` and `rowModesModel` (only works if `editMode="row"`).
+
+The `cellModesModel` prop accepts an object where the key corresponds to the row ID containing the cell whose mode will be changed.
```

Comment:
The problem is that this interface suggested is not TS and the code will not be properly formatted. The correct interface would be 

```ts
export type GridCellModesModelProps =
  | ({ mode: GridCellModes.View } & Omit<GridStopCellEditModeParams, 'id' | 'field'>)
  | ({ mode: GridCellModes.Edit } & Omit<GridStartCellEditModeParams, 'id' | 'field'>);

export type GridCellModesModel = Record<GridRowId, Record<string, GridCellModesModelProps>>;

export type GridRowModesModelProps =
  | ({ mode: GridRowModes.View } & Omit<GridStopRowEditModeParams, 'id' | 'field'>)
  | ({ mode: GridRowModes.Edit } & Omit<GridStartRowEditModeParams, 'id' | 'field'>);

export type GridRowModesModel = Record<GridRowId, GridRowModesModelProps>;
```

but it doesn't make any sense.

Reply:
Yeah. that interface is not really helpful in explaining it :).

Can we use jsx on this case to display the "template" object?

I think the paragraph describing the object is a bit convoluted. I tried simplifying it, but I don't think it's enough because the object itself is hard to explain while keeping the description accurate.

```suggestion
The `cellModesModel` prop accepts an object containing the `mode` (and additional options) for a given column field, in a given row, as in the example below.
The options accepted are the same available in apiRef.current.startCellEditMode and apiRef.current.stopCellEditMode.
```

Reply:
I applied the suggestion. We can only use JSX on JavaScript code. I think the example with the comments explain well the shape of the object.